### PR TITLE
Update third party dependencies

### PR DESCRIPTION
**Pull request by JanuszL**
- Google Benchmark 1.5.4 -> 1.5.5
- {fmt} 7.1.3- > 8.0.0
- GoogleTest 1.10.0 -> 1.11.0
- RapidJSON May 7 -> Jun 14
- Thrust 1.12.0 -> 1.12.1
- updates conda recipe

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It updates third party dependencies

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     Google Benchmark 1.5.4 -> 1.5.5
     {fmt} 7.1.3- > 8.0.0
     GoogleTest 1.10.0 -> 1.11.0
     RapidJSON May 7 -> Jun 14
     Thrust 1.12.0 -> 1.12.1
     updates conda recipe
     Affected modules and functionalities:
 - Key points relevant for the review:
     NA
 - Validation and testing:
     CI run
 - Documentation (including examples):
     NA

**JIRA TASK**: *[DALI-2167]*

### CODE REVIEW

File: conda/recipe/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # (C) Copyright IBM Corp. 2019. All Rights Reserved.
-# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2019 - 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (jantonguirao) [inline conda/recipe/meta.yaml:2]:
```suggestion
# Copyright (c) 2019-2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```

File: conda/recipe/meta.yaml [hunk 2]
```diff
@@ -64,12 +64,12 @@ requirements:
...
     - future >=0.17.1
-    - protobuf >=3.17.1
-    - libprotobuf-static >=3.17.1
+    - protobuf >=3.17.2
+    - libprotobuf-static >=3.17.2
     - libjpeg-turbo >=2.1.0
-    - tensorflow-gpu =2.2.0
-    - tensorflow-estimator =2.2.0
-    - tensorboard =2.2.2
+    - tensorflow-gpu =2.4.1
```
Comment (jantonguirao) [inline conda/recipe/meta.yaml:70]: do we force users to install TF?

**Review by jantonguirao — APPROVED** [review]

File: conda/recipe/meta.yaml (hunk 2 above)
Comment (JanuszL) [reply conda/recipe/meta.yaml:70]: Yes. We do. We need to prebuild plugin and have a TF version defined. Maybe in future we can think how to compile in fly during the installation (if that is possible at all).

File: conda/recipe/meta.yaml (hunk 1 above)
Comment (JanuszL) [reply conda/recipe/meta.yaml]: Done

**Review by szalpal — APPROVED** [review]

**Pull request merged**